Tests/MongoSwiftTests/ReadWriteConcernTests.swift

```diff
@@ -0,0 +1,218 @@
+    	coll3 = try db2.collection("coll3")
+    	expect(coll3.readConcern.level).to(equal("local"))
+
+    	// expect that a collection created from a DB with local RC can override the DB's RC
+    	var coll4 = try db2.createCollection("coll4", options: CreateCollectionOptions(readConcern: try ReadConcern(.majority)))
+    	expect(coll4.readConcern.level).to(equal("majority"))
+
+ 		// expect that a collection retrieved from a DB with local RC can override the DB's RC
+ 		coll4 = try db2.collection("coll4", options: CollectionOptions(readConcern: try ReadConcern(.majority)))
+    	expect(coll4.readConcern.level).to(equal("majority"))
+    }
+
+    func testOperationReadConcerns() throws {
+    	// setup a collection 
+    	let client = try MongoClient()
+    	let db = try client.db("test")
+    	defer { do { try db.drop() } catch {} }
+    	let coll = try db.createCollection("coll1")
+
+    	let command: Document = ["count": "coll1"]
+
+    	// run command with a valid readConcern
+    	let options1 = RunCommandOptions(readConcern: try ReadConcern(.local))
+    	let res1 = try db.runCommand(command, options: options1)
+        expect(res1["ok"] as? Double).to(equal(1.0))
+
+        // run command with an empty readConcern
+        let options2 = RunCommandOptions(readConcern: ReadConcern())
+        let res2 = try db.runCommand(command, options: options2)
+        expect(res2["ok"] as? Double).to(equal(1.0))
+
+        // running command with an invalid RC level should throw
+        let options3 = RunCommandOptions(readConcern: try ReadConcern("blah"))
+        expect(try db.runCommand(command, options: options3)).to(throwError())
+
+        // try various command + read concern pairs to make sure they work
+        expect(try coll.find(options: FindOptions(readConcern: try ReadConcern(.local)))).toNot(throwError())
+
+        expect(try coll.aggregate([["$project": ["a": 1] as Document]],
+        	options: AggregateOptions(readConcern: try ReadConcern(.majority)))).toNot(throwError())
+
+        expect(try coll.count(options: CountOptions(readConcern: try ReadConcern(.majority)))).toNot(throwError())
+
+        expect(try coll.distinct(fieldName: "a",
+        	options: DistinctOptions(readConcern: try ReadConcern(.local)))).toNot(throwError())
+    }
+
+    func testConnectionStrings() throws {
+    	let csPath = "\(self.getSpecsPath())/read-write-concern/tests/connection-string"
+    	let testFiles = try FileManager.default.contentsOfDirectory(atPath: csPath).filter { $0.hasSuffix(".json") }
+    	for filename in testFiles {
+    		let testFilePath = URL(fileURLWithPath: "\(csPath)/\(filename)")
+            let asDocument = try Document(fromJSONFile: testFilePath)
+            let tests: [Document] = try asDocument.get("tests")
+            for test in tests {
+	            let description: String = try test.get("description")
+                if description == "wtimeoutMS as an invalid number" { continue }
```

Comment:
I'm waiting to hear back from Kevin/Jesse about this case (see msgs in #mongoc) - it looks to me like the C driver just ignores a negative wtimeout value rather than throwing an error. but I don't see them skipping this case in their test runner for this so I'm not totally sure we should be skipping it

Reply:
to update, it looks like due a [bug](https://github.com/mongodb/mongo-c-driver/blob/master/tests/test-mongoc-read-write-concern.c#L86) in their test runner they stop executing after the first invalid case (`return` instead of `continue`), so this case wasn't actually being checked, and with that fix they don't pass it. so we will have to skip this until the C driver has a fix. will update with a C driver ticket number later on